File: RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.c
#include <stdlib.h>
#include <assert.h>
#include <stdio.h>

int removeDuplicates(int* nums, int numsSize) {
    int index = 0;
    int k = 0;
    int *aux = malloc(numsSize * sizeof(int));

    if (aux == NULL) {
        return -1;
    }

    for (int i = 0; i < numsSize; i++) {

        if (i == 0) {
            aux[index] = nums[i];
            k++;
        } else {
            if (nums[i] != aux[index]) {
                index++;
                aux[index] = nums[i];
                k++;
            }
        } 
    }

    for (int j = 0; j < numsSize; j++) {
        nums[j] = aux[j];
    }
    return k;
}

int main() {
    int nums[] = {0,0,1,1,1,2,2,3,3,4};
    int expectedNums[] = {0,1,2,3,4};
    int numsSize = 10;
    int expectedSize = 5;

    int k = removeDuplicates(nums, numsSize);

    assert(k == expectedSize);
    for (int i = 0; i < k; i++) {
        assert(nums[i] == expectedNums[i]);
    }

    for(int i = 0; i < 5; i++) {
        printf("\n%d", nums[i]);
    }

    printf("\n");
}

File: RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int index = 0;

        for(int j = 1; j < nums.length; j++) {
            if(nums[index] != nums[j]) {
                index++;
                nums[index] = nums[j];
            }
        }
        
        return index+1;
    }
}

public class RemoveDuplicatesFromSortedArray {
    public static void main(String[] args) {
        int[] nums = {0,0,1,1,1,2,2,3,3,4};
        int[] expectedNums = {0,1,2,3,4};

        Solution s = new Solution();
        int k = s.removeDuplicates(nums);

        assert k == expectedNums.length;
        for (int i = 0; i < k; i++) {
            assert nums[i] == expectedNums[i];
        }

        System.out.println("k = " + k);
        for (int i = 0; i < k; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}